File: src/AndroidXamarinBenchmark/AndroidXamarinBenchmark/Activities/ActivityCameraAccess.cs
using Android.App;
using Android.Content;
using Android.Graphics;
using Android.OS;
using Android.Provider;
using Android.Widget;

namespace AndroidXamarinBenchmark.Activities
{
    [Activity(Label = "ActivityCameraAccess")]
    public class ActivityCameraAccess : Activity
    {
        private static int CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE = 100;
        private ImageView imageView;

        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            SetContentView(Resource.Layout.ActivityCameraAccess);

            imageView = FindViewById<ImageView>(Resource.Id.imageViewPhoto);
            Button photoButton = FindViewById<Button>(Resource.Id.buttonTakePhoto);

            photoButton.Click += (sender, e) =>
            {
                Intent intent = new Intent(MediaStore.ActionImageCapture);
                StartActivityForResult(intent, CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE);
            };
        }

        protected override void OnActivityResult(int requestCode, Result resultCode, Intent data)
        {
            base.OnActivityResult(requestCode, resultCode, data);
            if (requestCode == CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE && resultCode == Result.Ok)
            {
                Bitmap photo = (Bitmap)data.Extras.Get("data");
                imageView.SetImageBitmap(photo);
            }
        }
    }
}

File: src/AndroidNativeBenchmark/app/src/main/java/com/rochapires/admin/androidnativebenchmark/Activities/ActivityCameraAccess.java
package com.rochapires.admin.androidnativebenchmark.Activities;

import android.content.Intent;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.provider.MediaStore;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.ImageView;
import com.rochapires.admin.androidnativebenchmark.R;

public class ActivityCameraAccess extends AppCompatActivity {

    private static final int CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE = 100;
    private ImageView imageView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_activity_camera_access);

        this.imageView = (ImageView)this.findViewById(R.id.imageViewPhoto);
    }

    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        if (requestCode == CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE && resultCode == RESULT_OK) {
            Bitmap photo = (Bitmap) data.getExtras().get("data");
            imageView.setImageBitmap(photo);
        }
    }

    public void startCameraIntent(View view) {
        Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
        startActivityForResult(intent, CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE);
    }
}
